sql/parser/walk.go

```diff
@@ -183,13 +189,33 @@ func WalkStmt(v Visitor, stmt Statement) {
 		if stmt.Where != nil {
 			stmt.Where.Expr = WalkExpr(v, stmt.Where.Expr)
 		}
-	case *Insert:
-		switch rows := stmt.Rows.(type) {
-		case Values:
-			for i, tuple := range rows {
-				rows[i] = WalkExpr(v, tuple).(Tuple)
+		for i, expr := range stmt.GroupBy {
+			stmt.GroupBy[i] = WalkExpr(v, expr)
+		}
+		if stmt.Having != nil {
+			stmt.Having.Expr = WalkExpr(v, stmt.Having.Expr)
+		}
+		for i, expr := range stmt.OrderBy {
+			stmt.OrderBy[i].Expr = WalkExpr(v, expr.Expr)
```

Comment:
Many databases do not allow substitution in `ORDER BY` or other clauses referencing columns (on the other hand, users want parameterized `ORDER BY`, and doing this outside the database's placeholder system is a common source of SQL injection). The idea is that placeholders are only for values; the skeleton of the query should be sufficient for constructing a query plan without filling in arguments (so this plan can be constructed once and reused for many sets of values).